driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala

```diff
@@ -83,18 +84,35 @@ case class StreamingContextService(policyStatusActor: Option[ActorRef] = None, g
     if (policyStatusActor.isDefined) {
       log.info(s"Listener added for: $policyId")
       policyStatusActor.get ? AddListener(policyId, (policyStatus: PolicyStatusModel, nodeCache: NodeCache) => {
-        if (policyStatus.status.id equals PolicyStatusEnum.Stopping.id) {
-          synchronized {
-            log.info("Stopping message received from Zookeeper")
-            SparkContextFactory.destroySparkStreamingContext
-            SparkContextFactory.destroySparkContext
-            policyStatusActor.get ! Kill (name)
-            policyStatusActor.get ? Update(PolicyStatusModel(policyId, PolicyStatusEnum.Stopped))
-            nodeCache.close()
-            if (exit) System.exit(0)
+        synchronized {
+          if (policyStatus.status.id equals PolicyStatusEnum.Stopping.id) {
+            try {
+
+              log.info("Stopping message received from Zookeeper")
+              SparkContextFactory.destroySparkStreamingContext
+              SparkContextFactory.destroySparkContext
+
+            } finally {
+
+              Try(Await.result(policyStatusActor.get ? Kill(name), timeout.duration) match {
+                case false => log.warn(s"The actor with name: $name are stopped previously")
```

Comment:
The actor with name: $name are stopped previously -> The actor with name: $name has been stopped previously